packages/overlay/src/overlay-stack.ts

```diff
@@ -152,7 +152,18 @@ export class OverlayStack {
         event.stopPropagation();
         event.preventDefault();
         await this.closeTopOverlay();
-        const target = document.elementFromPoint(event.clientX, event.clientY);
+        let target = document.elementFromPoint(event.clientX, event.clientY);
+        while (target && target.shadowRoot) {
+            const innerTarget = (
+                target.shadowRoot as unknown as {
+                    elementFromPoint: (x: number, y: number) => Element | null;
+                }
+            ).elementFromPoint(event.clientX, event.clientY);
+            if (!innerTarget || innerTarget === target) {
+                break;
+            }
+            target = innerTarget;
+        }
```

Comment:
Is this something other components might need to do? If so, we could extract it into a shared function.

Reply:
I'm not sure we do it anywhere else right now, but abstracting it as a helper will be useful longterm, yes!